packages/dashboard/src/pages/Settings/components/dialogs/RestoreDialog.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+    Tab,
+    experimentalStyled as styled,
+    tabClasses,
+    tabsClasses,
+    InputBase,
+    inputBaseClasses,
+} from '@material-ui/core'
+import { makeStyles } from '@material-ui/styles'
+import { useState } from 'react'
+import { useAsync } from 'react-use'
+import ConfirmDialog from '../../../../components/ConfirmDialog'
+import { MaskColorVar } from '@dimensiondev/maskbook-theme'
+import FileUpload from '../../../../components/FileUpload'
+import { Services } from '../../../../API'
+import BackupPreviewCard from '../BackupPreviewCard'
+import type { BackupPreview } from '../BackupPreviewCard'
+
+const useStyles = makeStyles(() => ({
+    container: { flex: 1 },
+    hide: { display: 'none' },
+}))
+
+const StyledTabList: typeof TabList = styled(TabList)(() => ({
+    [`& .${tabsClasses.indicator}`]: {
+        display: 'none',
+    },
+})) as any
+
+const StyledTab: typeof Tab = styled(Tab)(() => ({
+    [`&.${tabClasses.root}`]: {
+        flex: 1,
+        background: MaskColorVar.secondaryBackground,
+        textTransform: 'none',
+        [`&:first-of-type`]: {
+            borderTopLeftRadius: 4,
+            borderBottomLeftRadius: 4,
+        },
+        [`&:last-of-type`]: {
+            borderTopRightRadius: 4,
+            borderBottomRightRadius: 4,
+        },
+        [`&.${tabClasses.selected}`]: {
+            color: MaskColorVar.primaryContrastText,
+            background: MaskColorVar.primary,
+        },
+    },
+})) as any
+
+const StyledTabPanel = styled(TabPanel)(({ theme }) => ({
+    [`&.${tabPanelClasses.root}`]: {
+        marginTop: theme.spacing(2.5),
+        padding: 0,
+    },
+}))
+
+const TextArea = styled(InputBase)(({ theme }) => ({
+    [`&.${inputBaseClasses.root}`]: {
+        height: 180,
+        alignItems: 'flex-start',
+        background: MaskColorVar.secondaryBackground,
+        borderRadius: 8,
+        padding: theme.spacing(2),
+    },
+}))
+
+export interface RestoreDialogProps {
+    open: boolean
+    onClose(): void
+}
+
+export default function RestoreDialog({ open, onClose }: RestoreDialogProps) {
+    const classes = useStyles()
+    // tab switch
+    const [tab, setTab] = useState('file')
+    // paste text
+    const [text, setText] = useState('')
+    // file content
+    const [content, setContent] = useState('')
+    // parsed json
+    const [json, setJSON] = useState<BackupPreview | null>(null)
+    // backup id
+    const [id, setId] = useState('')
+
+    const handleClose = () => {
+        onClose()
+        setTab('file')
+        setText('')
+        setContent('')
+        setJSON(null)
+    }
+    const handleConfirm = async () => {
+        if (!json) return
+
+        await Services.Welcome.checkPermissionsAndRestore(id)
+    }
+
+    useAsync(async () => {
+        const str = tab === 'file' ? content : text
+
+        if (str) {
+            const obj = await Services.Welcome.parseBackupStr(str)
```

Comment:
`Services.Welcome.parseBackupStr` is not an async function

Reply:
No, any function in `Services` is async in runtime because they're RPC